pkg/mutation/mutation_function.go

```diff
@@ -25,6 +25,9 @@ func mutate(mutator types.Mutator, tester *path.Tester, obj *unstructured.Unstru
 type mutatorState struct {
 	mutator types.Mutator
 	tester  *path.Tester
+	// valueTest takes the input value and whether that value already existed.
+	// It returns true if the value should be mutated
+	valueTest func(interface{}, bool) bool
```

Comment:
Just curious - was this made pluggable to facilitate testing?

Reply:
For the AssignMetadata mutator, which currently has no value test b/c it already requires target values to be undefined.